natss/pkg/reconciler/dispatcher/natsschannel.go

```diff
@@ -111,13 +114,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return err
 	}
 
-	if !original.Status.IsReady() {
-		return fmt.Errorf("Channel is not ready. Cannot configure and update subscriber status")
-	}
-
 	// Don't modify the informers copy.
 	natssChannel := original.DeepCopy()
 
+	// See if the channel has been deleted.
+	if natssChannel.DeletionTimestamp != nil {
+		c := toChannel(natssChannel)
+
+		if _, err := r.natssDispatcher.UpdateSubscriptions(c, true); err != nil {
+			logging.FromContext(ctx).Error("Error updating subscriptions", zap.Any("channel", c), zap.Error(err))
+			return err
+		}
+		removeFinalizer(natssChannel)
+		_, err := r.NatssClientSet.MessagingV1alpha1().NatssChannels(natssChannel.Namespace).Update(natssChannel)
```

Comment:
Not directly related to your changes, but have you considered making `NatssClientSet` a `messagingv1alpha1.NatssChannelsGetter`? That would save you the effort of calling `r.NatssClientSet.MessagingV1alpha1().NatssChannels(ns)...` everywhere, and allow you yo use `r.NatssClientSet(ns)...`

Reply:
No I didn't, because I didn't focus on changing the existing code. You are right it would be shorter. Nevertheless I would like to not change it.

Reply:
Fair enough